### PR TITLE
Resolves #6: Season start & end date

### LINKED ISSUE #6: Season start & end date
- Add start & end date fields in season creation.
- Validate quest completion date entered during quest submission to only fall within these date range.

### PR DESCRIPTION
**Pull request by aronpadolina**

### CODE REVIEW

**Comment by nbcp** [comment]
needs rebase

**Comment by aronpadolina** [comment]
done with the rebase

File: src/app/ui/player-quest-list/submit-quest-dialog.component.html [hunk 1]
```diff
@@ -5,22 +5,23 @@ <h1 mat-dialog-title>
...
-      <input matInput disabled placeholder="Source" value="{{data.quest.source}}" />
+      <input matInput disabled placeholder="Source" value="{{data.quest.quest.source}}" />
     </mat-form-field>
     <br />
     <mat-form-field>
       <input matInput [matDatepicker]="picker" required [max]="currentDate" placeholder="Date Completed" formControlName="completed"/>
       <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
       <mat-datepicker #picker></mat-datepicker>
       <mat-error *ngIf="hasError('completed','required')">Completion date is a required field</mat-error>
+      <mat-error *ngIf="hasError('completed','valid')">Completion date should be within the seasons date range</mat-error>
```
Comment (nbcp) [inline src/app/ui/player-quest-list/submit-quest-dialog.component.html]: please add the actual start and end dates for the current season here so the user won't have to guess the valid dates

File: src/app/ui/player-quest-list/submit-quest-dialog.component.html [hunk 2]
```diff
@@ -5,22 +5,23 @@ <h1 mat-dialog-title>
 <form [formGroup]="form">
   <div mat-dialog-content>
     <mat-form-field class="content" [floatLabel]="true">
-      <input matInput disabled placeholder="Category" value="{{data.quest.category.name}}" />
+      <input matInput disabled placeholder="Category" value="{{data.quest.quest.category.name}}" />
```
Comment (nbcp) [inline src/app/ui/player-quest-list/submit-quest-dialog.component.html]: lets create a `quest: Quest` property and assign it with `data.quest.quest` OnInit so here in the template we can directly access  `{{quest.category.name}}`

File: src/app/ui/admin-page/admin-page.component.ts [hunk 3]
```diff
@@ -24,16 +29,18 @@ export class AdminPageComponent implements OnInit {
...
-    private seasonService: SeasonService
+    private seasonService: SeasonService,
+    private notifyService: NotifyService
   ) {}
 
   ngOnInit() {
     this.seasons$ = this.seasonService.getData();
   }
 
   addSeason(user: User) {
```
Comment (nbcp) [inline src/app/ui/admin-page/admin-page.component.ts]: this method can now be removed

Comment (aronpadolina) [reply src/app/ui/admin-page/admin-page.component.ts]: hi ben, which method should be removed? ngOnInit or addSeason or both?

File: src/app/ui/player-quest-list/submit-quest-dialog.component.html (hunk 1 above)
Comment (aronpadolina) [reply src/app/ui/player-quest-list/submit-quest-dialog.component.html]: actual start and end dates of the season is now added to the validation message

File: src/app/ui/player-quest-list/submit-quest-dialog.component.html (hunk 2 above)
Comment (aronpadolina) [reply src/app/ui/player-quest-list/submit-quest-dialog.component.html]: this is now implemented with my latest push

File: src/app/ui/admin-page/admin-page.component.ts (hunk 3 above)
Comment (aronpadolina) [reply src/app/ui/admin-page/admin-page.component.ts]: 'addSeason' method is now removed as well as other fields which are only used on this method

**Pull request merged**